app/models/host/managed.rb

```diff
@@ -828,6 +828,14 @@ def available_template_kinds(provisioning = nil)
     end.compact
   end
 
+  def templates_used
```

Comment:
this whole method can be written as one-liner
```ruby
Hash[available_template_kinds.map { |t| [t.template_kind_name, t.name] }]
```

but if you find it less readable, I don't insist